lib/state/simperium/middleware.ts

```diff
@@ -60,6 +92,13 @@ export const middleware: S.Middleware = store => {
       case 'REVISIONS_TOGGLE':
         fetchRevisions(store, nextState);
         break;
+
+      case 'TOGGLE_ANALYTICS_ENABLED':
+        setAnalyticsEnabled(nextState);
+        break;
+      case 'FETCH_PREFERENCES':
+        fetchPreferences(store);
```

Comment:
there should be no need to carry-over `fetchPreferences`

there's only a few cases we need to cover, even if it does take some care to really nail it

| | `state` | Display | User Action | App Response |
|---|---|---|---|---|
| Unfetched, changed by user | `null` | Indeterminate display but toggle-able | Toggles | Update state, middleware sends update to Simperium |
| Unfetched, received `index` or `update` | `null` | Flips to value of state | | `state` becomes `true` \| `false` |
| Fetched | `true` \| `false` | Normal toggle | Toggles | Update state, middleware sends update to Simperium |

What this boils down to is that there are really only two state: indeterminate and toggled. Either a remote update or a user action can set it into toggled mode. It can _only_ be indeterminate if we haven't heard from Simperium yet or for some strange reason there's no preference at all in the account.

Simperium will _always_ "fetch" the initial value and all we need to do is faithfully render that.